pkg/gardenlet/controller/federatedseed/extensions/shootstate_control.go

```diff
@@ -190,12 +214,40 @@ func purposeToString(purpose *string) string {
 	return *purpose
 }
 
-func fromRequest(req reconcile.Request) (clusterName string) {
+func (s *ShootStateControl) getExtensionObject(ctx context.Context, key types.NamespacedName, objectCreator func() client.Object) (extensionsv1alpha1.Object, error) {
+	obj := objectCreator()
+	if err := s.seedClient.Client().Get(ctx, key, obj); err != nil {
+		return nil, err
+	}
+	return apiextensions.Accessor(obj)
+}
+
+func (s *ShootStateControl) getShootState(ctx context.Context, req reconcile.Request) (*gardencorev1alpha1.ShootState, error) {
+	var clusterName string
 	if req.Namespace == "" {
 		// Handling for cluster-scoped backupentry extension resources.
 		clusterName, _ = common.ExtractShootDetailsFromBackupEntryName(req.Name)
 	} else {
 		clusterName = req.Namespace
 	}
-	return
+
+	cluster := &extensionsv1alpha1.Cluster{}
+	if err := s.seedClient.Client().Get(ctx, kutil.Key(clusterName), cluster); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, fmt.Errorf("could not get cluster with name %s : %v", clusterName, err)
+	}
+
+	shoot, err := s.shootRetriever.FromCluster(cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	shootState := &gardencorev1alpha1.ShootState{}
+	if err := s.k8sGardenClient.DirectClient().Get(ctx, kutil.Key(shoot.Namespace, shoot.Name), shootState); err != nil {
```

Comment:
This will retrieve the entire `ShootState` every time an extension object is changed (which can happen quite frequently).
```suggestion
	if err := s.k8sGardenClient.Client().Get(ctx, kutil.Key(shoot.Namespace, shoot.Name), shootState); err != nil {
```

Reply:
Ping, I think this one is not resolved.

Reply:
Sorry, must've reverted it when I removed the merge annotations.